Enable tests on 54L09 FLPR and 54L20 FLPR

nordic-segl opened the pull request:
Enable:
- samples/drivers/mbox,
- samples/drivers/watchdog,
- tests/boards/nrf/qdec (nRF54L20pdk only; no qdec on L09),
- tests/drivers/gpio/gpio_basic_api,
- tests/drivers/timer/nrf_grtc_timer,
- tests/drivers/uart/uart_elementary,
- tests/drivers/watchdog/wdt_basic_api.

on:
- nrf54l09pdk/nrf54l09/cpuflpr,
- nrf54l20pdk/nrf54l20/cpuflpr.

nordic-segl committed:
samples: drivers: mbox: Enable sample on L09 and L20

Add board overlays required to run mbox sample on
nRF54L09 and nRF54L20 targets.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl commented:
Depends on https://github.com/zephyrproject-rtos/zephyr/pull/79355

nordic-segl committed:
tests: drivers: gpio: gpio_basic_api: Enable test on nRF54L* cpuflpr

Add overlay required to run gpio_basic_api test on
- nrf54l09pdk/nrf54l09/cpuflpr and
- nrf54l20pdk/nrf54l20/cpuflpr.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

henrikbrixandersen commented on tests/drivers/gpio/gpio_basic_api/boards/nrf54l09pdk_nrf54l09_cpuflpr.overlay:
Did you mean to include the GPIO test changes in this PR?

nordic-segl replied:
Added third commit that enables zephyr/tests/drivers/timer/nrf_grtc_timer on FLPR core on L09 and L20.

henrikbrixandersen replied:
You should update the PR description then. As it is now, it is rather confusing.

nordic-segl replied:
Thanks, done.

nordic-segl committed:
tests: drivers: timer: Enable test on nRF54L* cpuflpr

Enable nrf_grtc_timer test on:
- nrf54l09pdk/nrf54l09/cpuflpr and
- nrf54l20pdk/nrf54l20/cpuflpr.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl committed:
tests: boards: nrf: qdec: Enable test on nRF54L20 FLPR

Add overlay required to run qdec tests
on nrf54l20pdk/nrf54l20/cpuflpr.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl committed:
tests: drivers: watchdog: wdt_basic_api: Enable test on nRF54L09 and L20

Enable wdt_basic_api test on:
- nrf54l09pdk/nrf54l09/cpuapp (missing supported),
- nrf54l09pdk/nrf54l09/cpuflpr (missing overlay),
- nrf54l20pdk/nrf54l20/cpuflpr (missing supported and overlay).

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl committed:
samples: drivers: watchdog: Enable sample on nrf54L09 and L20

Add overlays required to run watchdog sample on:
- nrf54l09pdk/nrf54l09/cpuapp,
- nrf54l09pdk/nrf54l09/cpuflpr,
- nrf54l20pdk/nrf54l20/cpuflpr.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl committed:
tests: drivers: uart: uart_elementary: Run test on nRF54L* FLPR

Add overlays required to run the test on:
- nrf54l09pdk/nrf54l09/cpuflpr,
- nrf54l20pdk/nrf54l20/cpuflpr.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl commented:
CI is green. Ready for review.

anangl commented on samples/drivers/mbox/sample.yaml:
```diff
@@ -71,9 +73,11 @@ tests:
-  sample.drivers.mbox.nrf54l15:
+  sample.drivers.mbox.nrf54l:
     platform_allow:
       - nrf54l15dk/nrf54l15/cpuapp
+      - nrf54l09pdk/nrf54l09/cpuapp
+      - nrf54l20pdk/nrf54l20/cpuapp
```
It would be nice to be consistent and use alphabetical order also when adding entries in this file (and in cmake files and the sysbuild one as well).

nordic-segl replied:
Thanks, I fixed the order.
